File: src/tauri/commands/sync.ts
import { safeInvoke } from "./_base";

export type SyncEntry = {
  project_code: string;
  project_name: string;
  task_name: string;
  process_code: string;
  category_label: string;
  hour: number;
  comment: string;
  regular_ot: number;
  midnight_ot: number;
};

export type SyncDailyReportRequest = {
  date: string;
  entries: SyncEntry[];
};

export type SyncResult = {
  success: boolean;
  message: string;
  synced_count: number;
  total_count: number;
};

export function syncDailyReport(request: SyncDailyReportRequest) {
  return safeInvoke<SyncResult>("sync_daily_report", { request });
}

File: src-tauri/src/models/sync.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct SyncDailyReportRequest {
    pub date: String,
    pub entries: Vec<SyncEntry>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct SyncEntry {
    pub project_code: String,
    pub project_name: String,
    pub task_name: String,
    pub process_code: String,
    pub category_label: String,
    pub hour: f64,
    pub comment: String,
    pub regular_ot: f64,
    pub midnight_ot: f64,
}

#[derive(Debug, Serialize)]
pub struct SyncResult {
    pub success: bool,
    pub message: String,
    pub synced_count: usize,
    pub total_count: usize,
}
